client/asset/eth/multirpc.go

```diff
@@ -569,12 +569,80 @@ func (m *multiRPCClient) voidUnusedNonce() {
 	m.lastNonce.stamp = time.Time{}
 }
 
-func (m *multiRPCClient) reconfigure(ctx context.Context, settings map[string]string) error {
+// createAndCheckProviders creates and connects to providers. It checks that
+// unknown providers have a sufficient api to trade and saves good providers to
+// file. One bad provider or connect problem will cause this to error.
+func createAndCheckProviders(walletDir string, endpoints []string, net dex.Network,
+	log dex.Logger, ctx context.Context) error {
+	var localCP map[string]bool
+	path := filepath.Join(walletDir, "compliant-providers.json")
+	b, err := os.ReadFile(path)
+	if err == nil {
+		if err := json.Unmarshal(b, &localCP); err != nil {
+			log.Warnf("Couldn't parse compliant providers file: %v", err)
+		}
+	}
```

Comment:
Looks like we are dropping the original error from `os.ReadFile`, maybe worth logging also (similar to how parsing error from `json.Unmarshal` is logged here) ?

Reply:
If it's something than os.ErrNotExists or whaterver

Reply:
Now printing the error with warn severity unless os.ErrNotExists